.gitlab/ci/on-push.yml

```diff
@@ -13,19 +13,17 @@ trigger-private-build:
   extends:
     - .default-job-settings
   rules:
-    - if: '$CI_PIPELINE_SOURCE =~ /^(push)$/ && $SKIP_PRIVATE_BUILD != "true"'
-    - if: '$NIGHTLY && $SKIP_PRIVATE_BUILD != "true"'
-    - if: '$INSTANCE_TESTS && $SKIP_PRIVATE_BUILD != "true"'
-    - if: '$DEMISTO_SDK_NIGHTLY && $SKIP_PRIVATE_BUILD != "true"'
+    - if: '$SKIP_PRIVATE_BUILD == "true"'
+      when: never
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push)$/'
+    - if: '$NIGHTLY'
+    - if: '$INSTANCE_TESTS'
+    - if: '$DEMISTO_SDK_NIGHTLY'
```

Comment:
Do we trigger a private build in these two cases currently? I didn't think we did for the `DEMISTO_SDK_NIGHTLY`.